torch_xla/core/dynamo_bridge.py

```diff
@@ -319,3 +311,67 @@ def optimized_mod(*args):
     return result
 
   return optimized_mod
+
+
+class FallBackNodeCollector(torch.fx.Interpreter):
+
+  def __init__(self, module):
+    super().__init__(module)
+    self._fallback_ops = []
+
+  def run_node(self, n: torch.fx.Node):
+    metrics.clear_counters()
+    result = super().run_node(n)
+    fallback_ops = get_fallback_ops()
+    if len(fallback_ops) > 0:
+      self._fallback_ops.append(n)
+    return result
+
+  def get_fallback_ops(self):
+    return self._fallback_ops
+
+
+def collect_inputs(model, *args):
+  model.xla_args = args
+  delattr(model, '_wrapped_call')
+  return model(*args)
+
+
+def extract_compiled_graph(xla_model, xla_args):
+  # execute model once to collect fallback ops
+  collector = FallBackNodeCollector(xla_model)
+  collector.run(*xla_args)
+  fallback_ops = collector.get_fallback_ops()
+
+  class XlaOperatorSupport(torch.fx.passes.operator_support.OperatorSupport):
+
+    def is_node_supported(self, submodules, node: torch.fx.Node) -> bool:
+      return node.op == "call_function" and (node not in fallback_ops or
+                                             node.target == operator.getitem)
+
+  # partition the model and exectue to collect inputs
+  supported_ops = XlaOperatorSupport()
+  partitioner = CapabilityBasedPartitioner(xla_model, supported_ops)
+  partitions = partitioner.propose_partitions()
+  partitioned_graph = partitioner.fuse_partitions(partitions)
+
+  for node in partitioned_graph.graph.nodes:
+    if node.op == "call_module" and "fused_" in node.name:
+      fused_module = getattr(partitioned_graph, node.name)
+      fused_module._wrapped_call = collect_inputs
+
+  partitioned_graph(*xla_args)
+
+  for node in partitioned_graph.graph.nodes:
+    if node.op == "call_module" and "fused_" in node.name:
+      fused_module = getattr(partitioned_graph, node.name)
+      partitioned_graph.delete_submodule(node.target)
+      with partitioned_graph.graph.inserting_after(node):
+        new_node = partitioned_graph.graph.call_function(
+            extract_internal(fused_module), node.args, None)
+        node.replace_all_uses_with(new_node)
+      partitioned_graph.graph.erase_node(node)
```

Comment:
Wondering why we need this `partitioned_graph.graph.erase_node(node)`? Can we do `graph.eliminate_dead_code()` like https://github.com/pytorch/pytorch/blob/0d66db1b2a9470a50d930308dbffda017500b80b/torch/_prims/nvfuser_executor.py#L465 or is this something different?